### PR TITLE
Upgrade go-ucfg to version 0.8.2

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Upgrades go-ucfg to version 0.8.2. (Currently this is pointing to master to get a full test run before releasing go-ucfg at version 0.8.2).

## Why is it important?

Includes the following fixes:

- Fixed nonzero validator to not fail on nil array or slice.
- Fixed nonzero validator to validate maps.
- Fixed required validator to validate maps.

## Checklist

- [X] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: metricbeat/mb/mb_test.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func TestModuleConfig(t *testing.T) {
 		{
 			name: "missing required field",
```
Comment (ycombinator) [inline metricbeat/mb/mb_test.go]: Maybe update the test name too?

**Comment by simitt** [comment]
@blakerouse do you have any objections against backporting this to `7.6`? There's a [bugfix in APM Server](https://github.com/elastic/apm-server/pull/3356) that should be backported to `7.6` relying on this fix.

**Comment by blakerouse** [comment]
@simitt I don't see a big issue with it being backported, but the 0.8.1 PR also would need to be backported.

**Comment by simitt** [comment]
Thanks for pointing that out @blakerouse. I talked to @urso about this, and since `0.8.1` would introduce a breaking change I'm not going to backport to `7.6`.

**Pull request merged**